src/rgw/rgw_notify.cc

```diff
@@ -2,17 +2,530 @@
+            if (set_min_marker(end_marker, entry.marker) < 0) {
+              ldout(cct, 1) << "ERROR: cannot determin minimum between malformed markers: " << end_marker << ", " << entry.marker << dendl;
+            } else {
+              ldout(cct, 20) << "INFO: end marker for removal: " << end_marker << dendl;
+            }
+            has_error = true;
+            return;
+          }
+        });
+        ++entry_idx;
+      }
+
+      // wait for all pending work to finish
+      waiter.async_wait(yield);
+
+      // delete all published entries from queue
+      if (remove_entries) {
+        librados::ObjectWriteOperation op;
+        cls_2pc_queue_remove_entries(op, end_marker); 
+        // TODO call operate async with yield
+        const auto ret = rados_ioctx.operate(queue_name, &op);
+        if (ret < 0) {
+          ldout(cct, 1) << "ERROR: failed to remove entries up to: " << end_marker  << " from queue: " 
+            << queue_name << ". error: " << ret << dendl;
+        } else {
+          ldout(cct, 20) << "INFO: removed entries up to: " << end_marker <<  " from queue: " 
+            << queue_name << dendl;
+        }
+      }
+      // TODO: cleanup expired reservations
+    }
+
+sched_next:
+    if (is_idle) {
+      boost::asio::deadline_timer timer(io_context);
+      timer.expires_from_now(boost::posix_time::microseconds(queue_idle_sleep_us));
+	    ldout(cct, 20) << "INFO: queue " << queue_name << " is idle, next processing will happen in: " << queue_idle_sleep_us << " usec" << dendl;
+      boost::system::error_code ec;
+	    timer.async_wait(yield[ec]);
+      // TODO check ec
+	    spawn::spawn(io_context, [this, &queue_name](spawn::yield_context yield) {
+          process_queue(yield, queue_name);
+          });
+    } else {
+	    ldout(cct, 20) << "INFO: queue " << queue_name << " is will be processed again. pending work for this queue: " << pending_work << dendl;
+	    spawn::spawn(io_context, [this, &queue_name](spawn::yield_context yield) {
+        process_queue(yield, queue_name);
+        });
+    }
+  }
+
+  // process all queues
+  // find which of the queues is owned by this daemon and process it
+  void process_queues(spawn::yield_context yield) {
+    auto has_error = false;
+    auto ret = try_to_create_queue_list();
+    if (ret < 0) {
+      // failed to create the queue list object
+      has_error = true;
+      goto sched_next;
+    }
+    {
+      queues_t queues;
+      ret = read_queue_list(queues);
+      if (ret < 0) {
+        // failed to create the queue list object
+        has_error = true;
+        goto sched_next;
+      }
+
+      for (const auto& queue_name : queues.list) {
+        // try to lock the queue to check if it is owned by this rgw
+        // or if ownershif needs to be taken
+        ret = rados::cls::lock::lock(&rados_ioctx, queue_name, queue_name+"_lock", 
```

Comment:
this is a synchronous call. you can use the `librados::ObjectWriteOperation` overload instead, and submit it with its yield context to `rgw_rados_operate()`

Reply:
this is happening when looking into the queue list - not the "fast path".
there are several other rados ops happening there, they should probably all move to the yielding api?